### DISCUSSION TITLE
What are CAGR range left/right?

### DISCUSSION THREAD

**Discussion by alexander-myltsev** [mbk-dev/okama#10] [Q&A] [answered]
Please, explain what are [target_cagr_range_left](https://github.com/mbk-dev/okama/blob/f43b0e0fcdc54bd7fa4fa36d087eb4e202b2ddbd/okama/frontier/multi_period.py#L381-L388) and [target_cagr_range_right](https://github.com/mbk-dev/okama/blob/f43b0e0fcdc54bd7fa4fa36d087eb4e202b2ddbd/okama/frontier/multi_period.py#L428-L441). I understand what CAGR is and why is there a range for each of them. I don't understand the meaning of `left` and `right`.

A link to the paper with an explanation is fine.

**Comment by chilango74** [comment, marked as answer]
Multi-period optimization is not convex so it is divided in 2 steps:
1.	From global CAGR min to global CAGR max. The objective function now is annualized standard deviation but can be any other risk metrics. The range of CAGR is defined by `target_cagr_range_left`.
2.	From local max CAGR (if exists) to the global max CAGR. `target_cagr_range_right` first look for assets to the right of the global max. Then determines corresponding CAGR range. The objective function here is also standard deviation but it looks for maximum values (not minimum as in the first step). The second step is optional.  Sometimes global max CAGR is one of the assets or `target_cagr_range_right` is too small. Then this step is skipped.

![image](https://user-images.githubusercontent.com/29829921/112248228-ace25900-8c66-11eb-8a9c-91eace4b24f6.png)
